app/repository/pg/user.go

```diff
@@ -5,17 +5,40 @@ package pg
 
 import (
 	"database/sql"
+	"fmt"
+	"log"
 
 	"github.com/go-devs-ua/octagon/app/entities"
 	"github.com/go-devs-ua/octagon/cfg"
+	_ "github.com/lib/pq"
 )
 
 // Repo wraps a database handle
-type Repo struct{ sql.DB }
+type Repo struct {
+	DB *sql.DB
+}
 
 // NewRepo will initialise new instance of Repo
 func NewRepo(opt cfg.Options) *Repo {
-	return &Repo{}
+	str := fmt.Sprintf("host=%v port=%v user=%v password=%v dbname=%v sslmode=disable",
+		opt.DBConfig.Host, opt.DBConfig.Port, opt.DBConfig.Username, opt.DBConfig.password, opt.DBConfig.DBName)
+
+	db, err := sql.Open("postgres", str)
```

Comment:
Repository MUST NOT create connection to the DB. 
It should come as an agrument to constructor and be stored in Repo struct DB field.

Reply:
Connection opening and DB ping should happen where we do all resources initialization. 
It looks like main is the right place.

Reply:
Fix in:
[fix: Call connection in main](https://github.com/go-devs-ua/octagon/pull/4/commits/1a77b437f50bc589577f71dbfbc25b553eb18e63)